lib/segment/src/index/field_index/map_index/immutable_map_index.rs

```diff
@@ -247,9 +250,23 @@ impl<N: MapIndexKey + ?Sized> ImmutableMapIndex<N> {
         Ok(result)
     }
 
-    pub fn check_values_any(&self, idx: PointOffsetType, check_fn: impl Fn(&N) -> bool) -> bool {
-        self.point_to_values
-            .check_values_any(idx, |v| check_fn(v.borrow()))
+    pub fn check_values_any(
+        &self,
+        idx: PointOffsetType,
+        hw_counter: &HardwareCounterCell,
+        check_fn: impl Fn(&N) -> bool,
+    ) -> bool {
+        // Overhead of accessing the index.
+        hw_counter
+            .payload_index_io_read_counter()
+            .incr_delta(MMAP_PTV_ACCESS_OVERHEAD);
+
+        self.point_to_values.check_values_any(idx, |v| {
+            let v: &N = v.borrow();
+            let size = <N as MmapValue>::mmapped_size((*v).as_referenced());
```

Comment:
Can't we get this size outside the hot path?

Reply:
Not really. If `N` is a `str`, it may always have a different size so either we do it here or iterate over the values twice.

Reply:
Any idea how expensive this could get?
Looking around I could not find a bench for the immutable map index unfortunately

Reply:
I haven't measured it directly but the size calculation is as I said unavoidable since `N` can be of type `str`.

We probably can improve performance a bit by using an integer for collecting the total size and accumulate this value into the HwCounter afterwards but I'd suggest to address this in a separate PR and focus on merging the PRs for now.